pkg/server/server.go

```diff
@@ -44,14 +50,110 @@ func (s *server) ListApplications(ctx context.Context, msg *pb.ListApplicationsR
 
 func (s *server) GetApplication(ctx context.Context, msg *pb.GetApplicationRequest) (*pb.GetApplicationResponse, error) {
 	app, err := s.kube.GetApplication(ctx, types.NamespacedName{Name: msg.Name, Namespace: msg.Namespace})
+	if err != nil {
+		return nil, fmt.Errorf("could not get application \"%s\": %w", app.Name, err)
+	}
 
+	src, automation, err := findFluxObjects(app)
 	if err != nil {
-		return nil, fmt.Errorf("could not get application: %s", err)
+		return nil, fmt.Errorf("could not get flux objects for application: %w", err)
+	}
+
+	name := types.NamespacedName{Name: app.Name, Namespace: app.Namespace}
+
+	if err := s.kube.GetResource(ctx, name, src); err != nil {
+		return nil, fmt.Errorf("could not get source for app %s: %w", app.Name, err)
+	}
+
+	if err := s.kube.GetResource(ctx, name, automation); err != nil {
+		return nil, fmt.Errorf("could not get automation for app %s: %w", app.Name, err)
+	}
+
+	// A Source is just an abstract interface, we need to get the underlying implementation.
+	var srcK8sConditions []metav1.Condition
+	switch st := src.(type) {
+	case *sourcev1.GitRepository:
+		srcK8sConditions = st.Status.Conditions
+	case *sourcev1.HelmRepository:
+		srcK8sConditions = st.Status.Conditions
+	}
+
+	srcConditions := mapConditions(srcK8sConditions)
+
+	var automationK8sConditions []metav1.Condition
+	switch at := automation.(type) {
+	case *kustomizev1.Kustomization:
+		automationK8sConditions = at.Status.Conditions
+	case *helmv2.HelmRelease:
+		automationK8sConditions = at.Status.Conditions
 	}
 
+	automationConditions := mapConditions(automationK8sConditions)
+
 	return &pb.GetApplicationResponse{Application: &pb.Application{
-		Name: app.Name,
-		Url:  app.Spec.URL,
-		Path: app.Spec.Path,
+		Name:                 app.Name,
+		Url:                  app.Spec.URL,
+		Path:                 app.Spec.Path,
+		SourceConditions:     srcConditions,
+		AutomationConditions: automationConditions,
 	}}, nil
 }
+
+// Returns k8s objects that can be used to find the cluster objects.
+// The first return argument is the source, the second is the automation
+func findFluxObjects(app *wego.Application) (client.Object, client.Object, error) {
+	st := app.Spec.SourceType
+	if st == "" {
+		// Apps that were created before the SourceType field exists will not have a SourceType defined.
+		// Assume git, since thats what the CLI defaults to.
+		st = wego.SourceTypeGit
+	}
+
```

Comment:
It's possible the only one of the objects exists when findFluxObjects is called and I think that should be OK.  If only the source object is present, we should still return it and vice versa for the deployment type.  In fact, neither may exist in the cluster yet and that should be OK.

Reply:
Changed to return `nil` with no error in the event of a "NotFound" error from kubernetes